### PR TITLE
Fixed fancy boxes not showing proper content amount.

### LINKED ISSUE #31762: Donut boxes go invisible if there's no donuts inside.
Take all but one donut out of the box. The box shows it's empty, but there's still a donut inside. Take it out, it goes invisible. The icon_state is down to -1 when it's empty for some reason.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 
:cl: DaxDupont
fix: Fancy boxes(ie: donut boxes) now show the proper content amount in the sprite and no longer go invisible when empty.
/:cl:

Fixes #31762

### CODE REVIEW

File: code/game/objects/items/storage/fancy.dm [hunk 1]
```diff
@@ -29,8 +29,7 @@
 
 /obj/item/storage/fancy/update_icon(itemremoved = 0)
 	if(fancy_open)
-		var/total_contents = src.contents.len - itemremoved
-		icon_state = "[icon_type]box[total_contents]"
+		icon_state = "[icon_type]box[src.contents.len]"
```
Comment (Cyberboss) [inline code/game/objects/items/storage/fancy.dm]: Remove the src

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
See my comment

File: code/game/objects/items/storage/fancy.dm (hunk 1 above)
Comment (DaxDupont) [reply code/game/objects/items/storage/fancy.dm]: Should I also delete the other src. like over here:
to_chat(user, "There are [contents.len <= 0 ? "no" : "[src.contents.len]"] [src.icon_type]s left.")

Comment (DaxDupont) [reply code/game/objects/items/storage/fancy.dm]: for content.len

Comment (Cyberboss) [reply code/game/objects/items/storage/fancy.dm]: sure if you want

**Review by Cyberboss — DISMISSED** [review]

File: code/game/objects/items/storage/fancy.dm [hunk 2]
```diff
@@ -27,20 +27,19 @@
 	for(var/i = 1 to storage_slots)
 		new spawn_type(src)
 
-/obj/item/storage/fancy/update_icon(itemremoved = 0)
+/obj/item/storage/fancy/update_icon()
```
Comment (optimumtact) [inline code/game/objects/items/storage/fancy.dm]: if you're going to remove this snowflake arg can you update the callers to this proc as well

Comment (optimumtact) [reply code/game/objects/items/storage/fancy.dm]: I'm pretty sure there is hack in the parent storage types removal proc for this

Comment (optimumtact) [reply code/game/objects/items/storage/fancy.dm]: addressed thanks.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**